go-runtime/ftl/call.go

```diff
@@ -19,37 +20,44 @@ func call[Req, Resp any](ctx context.Context, callee Ref, req Req) (resp Resp, e
 		return resp, fmt.Errorf("%s: failed to marshal request: %w", callee, err)
 	}
 
-	if overrider, ok := CallOverriderFromContext(ctx); ok {
-		override, uncheckedResp, err := overrider.OverrideCall(ctx, callee, req)
+	behavior, err := modulecontext.FromContext(ctx).BehaviorForVerb(modulecontext.Ref(callee))
+	if err != nil {
+		return resp, fmt.Errorf("%s: %w", callee, err)
+	}
+	switch behavior := behavior.(type) {
+	case modulecontext.MockBehavior:
```

Comment:
Instead of a switch in the `call` function with logic in each case, move the logic into the respective `VerbBehaviour` implementations.  A general pattern to watch for is that if significant logic around a "switch" is occurring outside the thing being switched on it might be time to encapsulate that logic into a method.